### DISCUSSION TITLE
[iOS] No sound output and gameplay stuck at begin after switch the game to background and back

### DISCUSSION THREAD

**Discussion by KrisCris** [ppy/osu#15886] [Q&A] [answered]
Device: iPad Pro 12.9 2021
System Version: 15.1
osu Build: 857e559f87931f4d8309bcf3a69f752f88b48f43

**issue:**

When I switch the game to the background and back, it lost sound output. And at this time, the gameplay will freeze at the beginning.

(I just have no idea how to open an issue... and it links me to here, weird)

**Comment by integerrr** [comment]
Sounds similar to #14729

**Reply by KrisCris** [reply]
Another issue is - switching output devices in the game settings will also cause the game to lose sound output

**Reply by frenzibyte** [reply]
That's reported in https://github.com/ppy/osu/discussions/15497.

**Comment by frenzibyte** [comment]
I cannot reproduce this on my iPhone 12.

Just for reference, can you reproduce this on all of the audio devices listed in the audio settings? (I assume what you have there is "Default" and "Voice")

**Reply by KrisCris** [reply]
I am trying hard but still cannot consistently reproduce this issue. But it does happen after it being switched to background or the device is locked. I am using the default output, cuz it’s impossible to have “voice” output any sound at all.

**Reply by KrisCris** [reply]
(Using the “voice” output would break the sound and I have to restart the game to get audio back…)

**Reply by KrisCris** [reply]
After watched the video from #14729, I can confirm that the behavior is almost the same when this error happens

**Reply by frenzibyte** [reply]
Okay one way I can reproduce this with a bluetooth device selected as the audio output, that can happen automatically if you have a bluetooth audio device paired to your phone. Would appreciate it if you can confirm that.

**Reply by KrisCris** [reply]
> Okay one way I can reproduce this with a bluetooth device selected as the audio output, that can happen automatically if you have a bluetooth audio device paired to your phone. Would appreciate it if you can confirm that.

I tried with my AirPods Pro but I can’t reproduce that.

**Reply by KrisCris** [reply]
Alright... I just realized the reason why I can't reproduce this issue when I wanted it to happen - I am using screen recording.
And I am finally able to reproduce this every time after I use my phone for recording lol.

Also, some weird behaviors are shown in the video...

https://user-images.githubusercontent.com/38860226/144539311-b213406b-c634-4b42-8005-c69ed58f3129.mp4

**Reply by frenzibyte** [reply]
Thanks, I've investigated through this and it appears to be suffering from the same underlying issue as https://github.com/ppy/osu/issues/14729, which is that audio device output get paused due to becoming unavailable but not resuming when it's available back.

Fixing this will likely require framework-side changes, and it may be a different fix from #14729 depending on the response to https://www.un4seen.com/forum/?topic=19601.msg137092#msg137092.

**Comment by smoogipoo** [comment, marked as answer]
Fixed by https://github.com/ppy/osu-framework/pull/4927.

**Reply by KrisCris** [reply]
Thanks, it works! @frenzibyte 

There is still a glitch that every time switching audio devices, the sound doesn't output until I click the volume button. Not a big problem, just let you know :)

**Reply by frenzibyte** [reply]
Good to hear. I can't reproduce the other issue you mentioned, but if you have solid reproduction steps, you can leave a separate discussion for that.

**Reply by KrisCris** [reply]
> Good to hear. I can't reproduce the other issue you mentioned, but if you have solid reproduction steps, you can leave a separate discussion for that.

ok, I will record a video for that. Basically, you switch between default and voice back and forth, and it will happen.

**Reply by peppy** [reply]
Sound not outputting is because your device is muted. Please un mute your device or use the volume button work around. That's how iOS works.

**Reply by KrisCris** [reply]
> Sound not outputting is because your device is muted. Please un mute your device or use the volume button work around. That's how iOS works.

My device is not muted. But I just realized it is not “No Sound Output”, but the sound is extremely low and seems are not using all speakers on my iPad. Lol
